src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -367,6 +367,19 @@ public function get_items( $request ) {
 			add_filter( 'post_password_required', array( $this, 'check_password_required' ), 10, 2 );
 		}
 
+		$post_type = get_post_type_object( $this->post_type );
+		if ( $post_type->hierarchical || 'attachment' === $this->post_type ) {
```

Comment:
This seems a bit backward. 

I'd loop through the results, check the same condition that prompts the get parent and prime based of that. It allows for theme and plugin developers making use if the inherit status away from attachments, hierarchical post types

```
foreach ( ... as $post ) {
  if ( 'inherit' === $post->post_status && $post->post_parent > 0 ) {
    $parent_ids[] = $post->post_parent;
  }
  // prime
}
```

Reply:
Agreed.

Reply:
I have removed the check, it is not needed. We need the post parent always, so need to check.